language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/Cek.hs

```diff
@@ -329,14 +350,21 @@ computeCek ctx env (Builtin ex bn) = do
   arity <- arityOf bn
   returnCek ctx (VBuiltin ex bn arity arity [] [] env)
 -- s ; ρ ▻ error A  ↦  <> A
-computeCek _ _  Error{} =
-    throwingWithCause _EvaluationError (UserEvaluationError CekEvaluationFailure) Nothing -- No value available for error
+computeCek _ _  err@Error{} =
+    throwingWithCause _EvaluationError (UserEvaluationError CekEvaluationFailure) $ Just err
 -- s ; ρ ▻ x  ↦  s ◅ ρ[ x ]
 computeCek ctx env (Var _ varName) = do
     spendBudget BVar (ExBudget 1 1) -- TODO
     val <- lookupVarName varName env
     returnCek ctx val
 
+-- | Call 'dischargeCekValue' over the received 'CekVal' and feed the resulting 'Term' to
+-- 'throwingWithCause' as the cause of the failure.
+throwingDischarged
```

Comment:
I'm on the fence over whether this is better than just calling `dischargeCekValue` inline, which I think is also pretty clear. Given a function `a -> b -> c` and a function `d -> b`, I'm not sure we need a specialized `a -> d -> c`.